frontends/web/src/components/aopp/aopp.tsx

```diff
@@ -0,0 +1,127 @@
+    public readonly state: State = {
+        aopp: {
+            state: 'inactive',
+            accounts: [],
+            errorCode: '',
+            address: '',
+        },
+    };
+
+    private chooseAccount = (code: AccountCode) => {
+        backendAPI.aoppChooseAccount(code);
+    }
+
+    public render(
+        { t, aopp }: RenderableProps<Props>,
+    ) {
+        if (!aopp) {
+            return null;
+        }
+        switch (aopp.state) {
+            case 'error':
+                return (
+                    <div>
+                        <p>{ t('aopp.addressRequested', { host: aopp.callbackHost }) }</p>
+                        <p>{ t(`error.${aopp.errorCode}`, { host: aopp.callbackHost }) }</p>
+                        <Button primary onclick={backendAPI.aoppCancel}>Dismiss</Button>
+                    </div>
+                );
+            case 'inactive':
+                // Inactive, waiting for action.
+                return null;
+            case 'awaiting-keystore':
+                return (
+                    <div>
+                        <p>{ t('aopp.addressRequested', { host: aopp.callbackHost }) }</p>
+                        <p>{ t('aopp.awaitingKeystore') }</p>
+                        <Button primary onclick={backendAPI.aoppCancel}>Cancel</Button>
+                    </div>
+                );
+            case 'choosing-account': {
+                const options = aopp.accounts.map(account => {
+                    return {
+                        text: account.name,
+                        value: account.code,
+                    };
+                });
+                return (
+                    <div>
+                        <Select
+                            options={[{
+                                text: t('buy.info.selectLabel'),
+                                disabled: true,
+                                value: 'choose',
+                            }, ...options]
+                            }
+                            defaultValue={'choose'}
+                            onChange={e => this.chooseAccount(e.target.value)}
+                            id="account"
+                        />
+                        <Button primary onclick={backendAPI.aoppCancel}>Cancel</Button>
+                    </div>
+                );
+            }
+            case 'syncing':
+                return <div>Syncing the account, please stand by.</div>;
+            case 'signing':
+                return (
+                    <div>
+                        <p>Address: {aopp.address}</p>
+                        <p>Please confirm on your BitBox.</p>
+                    </div>
+                );
+            case 'success':
+                return (
+                    <div>
+                        <p>Successfully delivered a fresh address to the third party.</p>
+                        <Button primary onclick={backendAPI.aoppCancel}>Dismiss</Button>
```

Comment:
There are a few strings to add to i18n and translate

Reply:
i left them literal because this is a prototype, maybe you will change everything anyway ;) i just put into i18n what i knew was gonna be needed for sure.